### ISSUE TITLE
Variable Learning Rate Matrix

### ISSUE THREAD

**Issue by githubaccount11** [pytorch/pytorch#33688]
## 🚀 Feature
Variable learning rate matrices option.

## Motivation

Many neural nets don't make full use of a computer's resources. Variable learning rate matrices could decrease time to reach optima while making use of otherwise wasted resources. Variable learning rate matrices are especially useful with attention based neural nets in case those are added in future versions.  A variable learning rate matrix could be as large as the matrix of weights or neurons ( perhaps larger if secondary learning rates were used ). It would be interesting to impose a super-intelligence  to control the values of the matrix.

## Pitch

I want to create an option for variable learning rate matrices. I will do all of the coding

**Comment by ngimel** [comment]
We need a more detailed description of the proposal.

**Comment by githubaccount11** [comment]
Is there any information you'd like to see in particular?

**Comment by ngimel** [comment]
From your issue it is absolutely unclear what you want to do. There is no description of what a variable learning rate matrix is, how it is useful, any references you can give?

**Comment by githubaccount11** [comment]
I'll probably just make it and if you guys like it you can use it too.

**Comment by vadimkantorov** [comment]
@ngimel I think the OP author refers to per-parameter adaptive learning rates. @githubaccount11 they are currently supported in some adaptive optimizers.

**Comment by githubaccount11** [comment]
Oh, then I should learn of those adaptive optimizer first. Should this issue now be closed?

**Comment by ngimel** [comment]
Yes, I think we can close this issue for now. In general, optimizers in pytorch are written in such a way that they can be easily extended in separate packages, and until a particular optimizer does not become mainstream, we typically don't take it into core package. So if you want to implement something new, you can do it in your own package and advertise your work on forums/slack.
